### PR TITLE
LPS-194062

### PR DESCRIPTION
**Pull request by icgsbr**
https://liferay.atlassian.net/browse/LPS-194062

### CODE REVIEW

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by eltonccdantas** [comment]
🕵🏻

**Comment by eltonccdantas** [comment]
Hey @icgsbr, I've analyzed your pull request and found two failures that might be related to it, one is related to `semantic-versioning` and another one is related to `com.liferay.portal.osgi.debug.declarative.service.test.DeclarativeServiceDependencyManagerTest` and carries the error message below:

```
org.junit.ComparisonFailure: expected:<[All declarative service components are satisfied]> but was:<[
	Bundle {id: 835, name: com.liferay.commerce.product.definitions.web, version: 7.0.114}
		Declarative Service {id: 9143, name: com.liferay.commerce.product.definitions.web.internal.application.list.CPDefinitionsPanelApp, unsatisfied references: 
			{name: _portlet, target: (javax.portlet.name=com_liferay_commerce_product_definitions_web_internal_portlet_CPDefinitionsPortlet)}
		}]>
		at org.junit.Assert.assertEquals(Assert.java:117)
		at org.junit.Assert.assertEquals(Assert.java:146)
		at com.liferay.portal.osgi.debug.declarative.service.test.DeclarativeServiceDependencyManagerTest.testDeclarativeServiceDependencyManagerResolvedDependencies(DeclarativeServiceDependencyManagerTest.java:109)
		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:344)
		at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
		at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
		at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
```
Would you be able to take a closer look at them, rebase and trigger `ci:test:relevant` again?

[Comparison run ](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=2&testrayRunIdB=2414810043&testrayRunIdA=2413431560&statusB=3)

**Comment by eltonccdantas** [comment]
I'll trigger `ci:test:relevant` again so we can double-check if the failures persist on a second run.

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by eltonccdantas** [comment]
Hey @icgsbr, good news... the failures seem to be unrelated to your PR. Is any other dev review needed?


[comparison run](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1693392313314&testrayRunIdA=2414574057&testrayRunIdB=2415126434)

**Comment by eltonccdantas** [comment]
> Hi @eltonccdantas !
> 
> I think it's prudent to ask for an additional dev review; specially one from peds.
> 
> I'm also doing some local tests aswell and the fix seems to be in order.

Perfect, I'll request it in our channel.

**Comment by eltonccdantas** [comment]
Hey @icgsbr, since it involves `Forms` as well, @matuzalemsteles and I think it might be a good idea to have it reviewed by a developer from BPM.

**Comment by pedro-michel** [comment]
Agreed with @eltonccdantas .

**Comment by eltonccdantas** [comment]
Hey @icgsbr, is there anything I can do from a QA perspective to help you work on this PR?

**Comment by icgsbr** [comment]
Hi @eltonccdantas !

I don't think so. I'm just waiting for a confirmation on how the review dynamic is gonna work, since there are two componenets related.

File: modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/helper/DDMFormRendererHelper.java
Comment (matuzalemsteles) [inline modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/helper/DDMFormRendererHelper.java:139]: This validation maybe shouldn't be done in the `ConfigurationModelToDDMFormValuesConverter` class? What do you think @ethib137 ?

Comment (icgsbr) [reply modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/helper/DDMFormRendererHelper.java:139]: Hi @matuzalemsteles, for what I've noticed, when reproducing the bug, the code doesn't stop at `ConfigurationModelToDDMFormValuesConverter`, furthermore, the mentioned class doesn't have the required elements for the validation.

Comment (icgsbr) [reply modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/helper/DDMFormRendererHelper.java]: Done!

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:relevant

**Review by thiago-buarqque — APPROVED** [review]
The code looks good to me. Do we still need the BPM developer review mentioned [here](https://github.com/liferay-peds/liferay-portal/pull/227#issuecomment-1699697362)?

**Comment by icgsbr** [comment]
As far as I know we do not need @thiago-buarqque

**Comment by thiago-buarqque** [comment]
Great, I guess you can forward it to Brian once everything is okay with the CI. If you're in a rush just make sure the errors (in the previous CI, the current one seems to be broken related to the Github/Jenkins API) are not related to your changes and send it manually to Brian.

**Comment by ethib137** [comment]
Hey guys, please do still get a BPM review for this issue. They may be aware of another pattern for doing this already, so we will be benefited by having them look at this as well.

**Comment by thiago-buarqque** [comment]
I'll find someone from BPM to take a look at this

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by thiago-buarqque** [comment]
Hey @icgsbr, could you forward this to [liferay-forms](https://github.com/liferay-forms/liferay-portal/pulls) so it can be reviewed by someone from BPM?

**Comment by icgsbr** [comment]
Hi @thiago-buarqque , I talked to the BPM team and they said to just mention them on this PR. @liferay-forms

**Comment by thiago-buarqque** [comment]
That's weird, someone else from BPM said to forward there.

**Comment by icgsbr** [comment]
ci:test:stable

File: modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/util/ConfigurationModelToDDMFormValuesConverter.java
Comment (marcelabc) [inline modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/util/ConfigurationModelToDDMFormValuesConverter.java:224]: It is not good to chain methods (configurationOverrideProperties.get().toString()).
I thought about two options (but there is more):

1. GetterUtil.getString(configurationOverrideProperties.get(ddmFormFieldValue.getName()))
2. MapUtil.getString(configurationOverrideProperties, ddmFormFieldValue.getName())

File: modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/util/ConfigurationModelToDDMFormValuesConverter.java
Comment (marcelabc) [inline modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/util/ConfigurationModelToDDMFormValuesConverter.java:206]: I would rename to:

1.  _validateDDMFormValuesWithConfigurationOverrideProperties()
2. _validateDDMFormValuesWithOverrideProperties()

File: modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/util/ConfigurationModelToDDMFormValuesConverter.java
Comment (marcelabc) [inline modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/util/ConfigurationModelToDDMFormValuesConverter.java:209]: Is it working to declare this variable outside the loop?
Because I see that `localizedValue` (Map) will get "bigger" because of `localizedValue.addString()`. This method is adding (`put()`) more values into this Map. Then, this Map is being added to each `ddmFormFieldValue` with different values.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by icgsbr** [comment]
Thanks for the review @marcelabc ! I'll take a look on everthing you pointed out! :)

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by john-co** [comment]
hi @icgsbr, do you need a follow up review from bpm team?

**Comment by icgsbr** [comment]
Hi @john-co ! Since the requested changes were done, all makes me believe that we're good to go.

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**
